### PR TITLE
Populate origin.user attribute from the SAN field of client cert

### PR DESCRIPTION
**Pull request by wattli**

### CODE REVIEW

File: src/envoy/mixer/http_control.cc [hunk 1]
```diff
@@ -169,9 +171,12 @@ void HttpControl::FillCheckAttributes(HeaderMap& header_map, Attributes* attr) {
   }
 }
 
-void HttpControl::Check(HttpRequestDataPtr request_data, HeaderMap& headers,
+void HttpControl::Check(HttpRequestDataPtr request_data, HeaderMap& headers, std::string origin_user,
                         DoneFunc on_done) {
   FillCheckAttributes(headers, &request_data->attributes);
+  if (origin_user != "") {
```
Comment (qiwzhang) [inline src/envoy/mixer/http_control.cc]: you can call SetStringAttribute

File: src/envoy/mixer/http_filter.cc [hunk 2]
```diff
@@ -151,8 +152,15 @@ class Instance : public Http::StreamFilter, public Http::AccessLog::Instance {
     state_ = Calling;
     initiating_call_ = true;
     request_data_ = std::make_shared<HttpRequestData>();
+
+    std::string origin_user;
+    Ssl::Connection* ssl = const_cast<Ssl::Connection *>(decoder_callbacks_->ssl());
+    if (ssl != nullptr) {
+      origin_user = ssl->uriSanPeerCertificate();
```
Comment (qiwzhang) [inline src/envoy/mixer/http_filter.cc]: can we use const for this function?  or why do we need const_cast?

File: src/envoy/mixer/http_control.cc (hunk 1 above)
Comment (wattli) [inline src/envoy/mixer/http_control.cc]: Done

File: src/envoy/mixer/http_filter.cc (hunk 2 above)
Comment (wattli) [inline src/envoy/mixer/http_filter.cc]: Yes, it's something i can add to envoy.

File: src/envoy/mixer/http_control.cc [hunk 3]
```diff
@@ -169,9 +171,10 @@ void HttpControl::FillCheckAttributes(HeaderMap& header_map, Attributes* attr) {
   }
 }
 
-void HttpControl::Check(HttpRequestDataPtr request_data, HeaderMap& headers,
+void HttpControl::Check(HttpRequestDataPtr request_data, HeaderMap& headers, std::string origin_user,
```
Comment (qiwzhang) [inline src/envoy/mixer/http_control.cc]: it seems that format is wrong.

Comment (wattli) [inline src/envoy/mixer/http_control.cc]: Done

**Pull request merged**